core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -1564,6 +1564,14 @@ class SparkContext(config: SparkConf) extends Logging with ExecutorAllocationCli
     }
   }
 
+  def killTasks(tasks: HashSet[Long], taskInfo: HashMap[Long, TaskInfo]): Boolean = {
```

Comment:
It is not suitable to add a public method here in `SparkContext`, `SparkContext` is a public entry point, any method adds to here should be considered carefully.  In your case looks like only Spark internally will use this method, why not directly change the `TaskSetManager`?